### PR TITLE
Fix 'empty flash' due to delayed rendering of framework components.

### PR DESCRIPTION
**Pull request by TomTirapani**
Don't trigger Grid.dataReaction until the ColumnHeaders have been rendered. This allows AgGrid to keep the loading overlay displayed until framework components have been rendered, and prevents a 'flash' that can occur during grid initialisation.

You can see this in action by opening the 'Complex Docked View' with CPU 6x slowdown, and comparing against the same in https://toolbox-dev.xh.io/app/containers/dock

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [ ] Caught up with `develop` branch as of last change.
- [ ] Added CHANGELOG entry, or determined not required.
- [ ] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [ ] Updated doc comments / prop-types, or determined not required.
- [ ] Reviewed and tested on Mobile, or determined not required.
- [ ] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: cmp/grid/GridModel.js [hunk 1]
```diff
@@ -697,6 +699,17 @@ export class GridModel {
         if (colIds.length) this.agColumnApi.autoSizeColumns(colIds);
     }
```
Comment (lbwexler) [inline cmp/grid/GridModel.js:701]: A bit unfortunate that we don't have a local implementation model where we can hang implementation state like this.

**Comment by lbwexler** [comment]
Seems like a clever workaround.  Wondering if there is a way we can introduce a LocalGridModel to keep this implementation state?  Might be useful going forward to adopt that pattern of seperating the API model from any implementation related state.

File: cmp/grid/GridModel.js [hunk 2]
```diff
@@ -697,6 +699,17 @@ export class GridModel {
...
 
+    /**
+     * Set a flag acknowledging that a framework component has been mounted.
+     * We debounce here catch the end of a collection of components mounted at
+     * the same time (i.e. column header initialisation)
+     */
+    @debounced(50)
+    @action
+    noteFrameworkCmpMounted() {
+        this.frameworkCmpsMounted = true;
```
Comment (lbwexler) [inline cmp/grid/GridModel.js:710]: why do we want/need to wait/delay until they are all mounted?  Doesn't this just become a no-op for the subsequent calls?

**Comment by TomTirapani** [comment]
Thanks for taking a look Lee - push a couple of cleanups.

Regarding the local model - Grid actually already has a LocalModel, which I agree would be a better candidate to keep this state. The problem is making the Grid's LocalModel accessible to the ColumnHeader. We may be able to put a reference to it on the GridModel and access it that way, but then its no longer really a LocalModel...

**Comment by lbwexler** [comment]
Tom and I just had a convo about this -- we'll move the workaround to the empty text component we already have in the local model, and thereby isolate the fix a bit more to the grid implementation.  
Conceptually it makes sense to have that component ready before rendering!  Even before this agGrid version I had noticed that you would get the default text rendering for a bit even if you had turned off emptyText -- this will fix that.

**Pull request merged**